packages/jaeger-ui/src/model/ddg/transformDdgData.tsx

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {
+  PathElem,
+  TDdgModel,
+  TDdgPayload,
+  TDdgPayloadEntry,
+  TDdgPath,
+  TDdgPathElemsByDistance,
+  TDdgServiceMap,
+  TDdgVisibilityIdxToPathElem,
+} from './types';
+
+const stringifyPayloadEntry = ({ service, operation }: TDdgPayloadEntry) => `${service}::${operation}`;
+
+export default function transformDdgData(
+  payload: TDdgPayload,
+  { service: focalService, operation: focalOperation }: { service: string; operation?: string }
+): TDdgModel {
+  const serviceMap: TDdgServiceMap = new Map();
+  const pathElemsByDistance: TDdgPathElemsByDistance = new Map();
+
+  const paths = payload
+    .slice()
+    .sort((a, b) =>
+      a
+        .map(stringifyPayloadEntry)
+        .join()
+        .localeCompare(b.map(stringifyPayloadEntry).join())
```

Comment:
I added a note in another thread, but `localeCompare` has very poor performance: https://jsperf.com/localecompare/38

Reply:
Also, you might consider doing the stringify and join once, up-front. For large sets of paths, this might be quite a hit on perf.

Reply:
I put in a map to avoid `path.map(stringifyPayloadEntry).join()` on the same path twice. And changed from `a.localeCompare(b)` to `a > b ? 1 : -1`